geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteFunctionNoAckOp.java

```diff
@@ -236,4 +236,16 @@ protected Message createResponseMessage() {
     }
   }
 
+  private static List<ServerLocation> getAllServers(PoolImpl pool) {
+    List<ServerLocation> servers;
+    if (pool.getLocators() == null || pool.getLocators().isEmpty()) {
+      servers = ((ExplicitConnectionSourceImpl) pool.getConnectionSource()).getAllServers();
```

Comment:
It seems like getAllServers ought to be a method on ConnectionSource, so this code doesn't have to check for the different types of connection sources and cast.